q2_types/tabular/tests/test_transformers.py

```diff
@@ -68,5 +73,322 @@ def test_jsonl_to_metadata(self):
                                        'faithpd_refdist.table.jsonl')
         _, exp = self.transform_format(TableJSONLFileFormat, pd.DataFrame,
                                        'faithpd_refdist.table.jsonl')
+
         exp = exp.set_index('id')
-        pd.testing.assert_frame_equal(obs.to_dataframe(), exp)
+
+        obs = obs.to_dataframe()
+        obs.index = obs.index.astype('string')
+        obs['group'] = obs['group'].astype('string')
+
+        pd.testing.assert_frame_equal(obs, exp)
+
+
+class TestDataframeToJsonlTypeHandling(TestPluginBase):
```

Comment:
Could we add a test for the JSON representation of each of these types. We're kind of relying on the implementation of to_json and from_json of pandas, which is fine, but it leaves what happens in the middle a little under-specified, which isn't great for a file-format and couples us to whatever pandas decided to do this week.

e.g. I would like to know that a `date` column comes out as `{"my_col": "2025-06-02"}` and not `{"my_col": "2025-06-02T00:00:00"}` (which pandas wouldn't have any problem round-tripping, but which would be weird).